java/rocksjni/rocksjni.cc

```diff
@@ -2277,10 +2277,11 @@ jlongArray Java_org_rocksdb_RocksDB_getApproximateSizes(
 
   auto ranges = std::unique_ptr<ROCKSDB_NAMESPACE::Range[]>(
       new ROCKSDB_NAMESPACE::Range[range_count]);
+  size_t range_offset = 0;
   for (jsize i = 0; i < jlen; ++i) {
     auto* start = reinterpret_cast<ROCKSDB_NAMESPACE::Slice*>(jranges[i]);
     auto* limit = reinterpret_cast<ROCKSDB_NAMESPACE::Slice*>(jranges[++i]);
-    ranges.get()[i] = ROCKSDB_NAMESPACE::Range(*start, *limit);
+    ranges.get()[range_offset++] = ROCKSDB_NAMESPACE::Range(*start, *limit);
```

Comment:
So I think that on x64 (macOS):

1. `size_t` is `uint64_t`
2. `jsize` is `int`, which is 32 bits.

So I don't quite understand the change for this issue.... As previously `i` could only be 32 bits, why does increasing it to 64 bits fix this, when the max could only ever be 32 bits.

Reply:
Issue is that i variable is increasing two times faster thus code writes after allocated memory.

Reply:
Doh! Of course!